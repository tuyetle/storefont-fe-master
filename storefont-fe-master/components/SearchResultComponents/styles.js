import variables from '~/styles/variables';
import { css } from 'glamor';

export default {
  content: css({
    display: 'flex',
    flexFlow: 'row nowrap',
    flex: '1 1 auto',
    position: 'relative',
    zIndex: '1',
    height: variables.$mobileSearchingResultHeight,
    '@media (min-width: 768px)': {
      height: variables.$searchingResultHeight,
    },
  }),
  searchBar: css({
    position: 'relative',
    zIndex: '10',
    top: '0',
  }),
  listingView: css({
    display: 'flex',
    flexFlow: 'column nowrap',
    flex: '1 1 50%',
    minWidth: '0',
    width: '100vw',
    height: '100%',
  }),
  mapView: css({
    flexBasis: '57%',
    top: 'initial',
    right: 'initial',
    bottom: 'initial',
    left: 'initial',
    borderLeft: `0.1em solid ${variables.$searchResultMapBorder}`,
    flexGrow: '0',
    flexShrink: '0',
    backgroundColor: variables.$searchResultMapBackground,

    '&.hide': {
      flexBasis: '0',
    },
  }),
  header: css({
    height: '10%',
    backgroundColor: variables.$searchResultBarBackgroundColor,
    '@media (min-width: 768px)': {
      height: '2em',
    },
  }),
  actionBar: css({
    flex: '0 0 auto',
    width: '100%',
    height: '100%',
    color: variables.$searchResultActionBarColor,
    backgroundColor: variables.$searchResultBarBackgroundColor,
    boxSizing: 'border-box',
    paddingLeft: '0.9em',
    paddingRight: '0.7em',

    '@media (min-width: 768px)': {
      display: 'flex',
      flexFlow: 'row nowrap',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingLeft: '1.875em',
      paddingRight: '1.875em',
    },
  }),
  results: css({
    fontSize: '0.8125em',
    display: 'inline-block',
    verticalAlign: 'middle',
    color: 'darkslategray',
    marginTop: '1.2em',
    '@media (min-width: 768px)': {
      display: 'inline-flex',
      flexDirection: 'row',
      marginLeft: '0',
      marginTop: '0',
      fontSize: '0.9375em',
    },
  }),
  buttons: css({
    display: 'inline-block',
    verticalAlign: 'middle',
    float: 'right',
    marginTop: '0.4em',

    '& .form-group': {
      marginBottom: '0',
      '& .btn': {
        marginBottom: '0',
        padding: '0.375em',
      },
    },
    '@media (min-width: 768px)': {
      justifyContent: 'flex-end',
      flex: '0 1 auto',
      float: 'none',
    },
  }),
  mainView: css({
    flex: '1 1 auto',
    backgroundColor: variables.$searchResultBarBackgroundColor,
  }),
  mainGirdMapView: css({
    height: 'calc(100% - 6em)',
    overflowX: 'hidden',
    overflowY: 'scroll',
  }),
  mainViewTranscluder: css({
    height: '100%',
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    marginTop: '0',
    '& h6': {
      fontWeight: 'normal',
      fontSize: '0.9em',
    },
    '& h5': {
      fontSize: '1.25em',
    },
    '@media (min-width: 768px)': {
      marginTop: '0.4em',
    },

  }),
  footer: css({
    backgroundColor: variables.$searchResultBarBackgroundColor,
    padding: '1em',
    height: '4em',
  }),
  number: css({
    fontWeight: '600',
    margin: '0 0.4em',
  }),
};
