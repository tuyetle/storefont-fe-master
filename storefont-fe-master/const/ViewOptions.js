const ViewOptions = {
  Grid: 'Grid',
  List: 'List',
  Map: 'Map',
};

export default ViewOptions;
