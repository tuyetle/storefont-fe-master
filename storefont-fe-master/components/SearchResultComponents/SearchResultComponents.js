import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'lodash/noop';
import { NotMobile, Mobile } from '~/config/media';
import { translate } from 'react-i18next';
import GridListSwitcher from '~/components/SearchResultComponents/GridListSwitcher/GridListSwitcher';
import PaginationComponent from '~/components/SearchResultComponents/Pagination/Pagination';
import GoogleMapSearch from '~/components/GoogleMapSearch/GoogleMapSearch';
import SortBarBox from '~/components/SearchResultComponents/SortBarBox/SortBarBox';
import ResultGridView from '~/components/SearchResultComponents/ResultGridView/ResultGridView';
import ResultListView from '~/components/SearchResultComponents/ResultListView/ResultListView';
import ViewOptions from '~/const/ViewOptions';
import styles from './styles';

const renderSearchResults = (
  listings, onListingSelected, onListingSaved,
  viewMode, showMap, facetSearchItems,
) => {
  let mainViewClass = `${styles.mainView}`;
  if (showMap && viewMode === ViewOptions.Grid) {
    mainViewClass += ` ${styles.mainGirdMapView}`;
  }

  return (
    <div className={`${mainViewClass}`}>
      <div className={`${styles.mainViewTranscluder}`}>
        {
          viewMode === ViewOptions.Grid &&
          <ResultGridView
            listings={listings}
            showMap={showMap}
            onListingSelected={onListingSelected}
            onListingSaved={onListingSaved}
          />
        }
        {
          viewMode === ViewOptions.List &&
          <ResultListView
            listings={listings}
            showMap={showMap}
            facetSearchItems={facetSearchItems}
            onListingSelected={onListingSelected}
            onListingSaved={onListingSaved}
          />
        }
      </div>
    </div>
  );
};

const SearchResultComponents = ({
  t,
  listings, listingsMapResult, foundItems, pageSize, listingsOfGroup, onMarkerGroupListingClick,
  currentPage, activeListing, showMap, onViewModeChanged, onMapViewModeChanged, onListingSelected,
  onListingSaved, onPageChanged, onActiveListing, activeListingResults, facetSearchItems,
  viewMode, onSortRusultChanged,
}) => (
  <div className={`${styles.content}`}>
    <NotMobile>
      <div className={`${styles.listingView}`}>
        <div className={`${styles.header}`}>
          <div className={`${styles.actionBar}`}>
            <div className={`${styles.results}`}>
              {t('searchResult:have')} <strong className={`${styles.number}`}>{foundItems}</strong> {t('searchResult:result')}
            </div>
            <div className={`${styles.buttons}`}>
              <SortBarBox onSortRusultChanged={onSortRusultChanged} />
              <GridListSwitcher
                changeModeViewHandler={onViewModeChanged}
                changeMapViewModeHandler={onMapViewModeChanged}
              />
            </div>
          </div>
        </div>
        {
          renderSearchResults(
            listings, onListingSelected, onListingSaved,
            viewMode, showMap, facetSearchItems,
          )
        }
        <div className={`${styles.footer}`}>
          <PaginationComponent
            onClickHandler={onPageChanged}
            pageSize={pageSize}
            currentPage={currentPage}
            total={foundItems}
          />
        </div>
      </div>
      {showMap ? (
        <div className={`${styles.mapView}`}>
          <GoogleMapSearch
            markersData={listingsMapResult}
            onActiveListing={onActiveListing}
            currentActive={activeListing}
            activeListing={activeListingResults}
            onListingSelected={onListingSelected}
          />
        </div>
      ) : <div />}
    </NotMobile>
    <Mobile>
      <div>
        {!showMap ? (
          <div className={`${styles.listingView}`}>
            <div className={`${styles.header}`}>
              <div className={`${styles.actionBar}`}>
                <SortBarBox onSortRusultChanged={onSortRusultChanged} />
                <div className={`${styles.results}`}>
                  {t('searchResult:have')} <strong className={`${styles.number}`}>{foundItems}</strong>  {t('searchResult:result')}
                </div>
                <div className={`${styles.buttons}`}>
                  <GridListSwitcher
                    changeModeViewHandler={onViewModeChanged}
                    changeMapViewModeHandler={onMapViewModeChanged}
                  />
                </div>
              </div>
            </div>
            {renderSearchResults(
              listings, onListingSelected, onListingSaved,
              viewMode, showMap, facetSearchItems,
            )}
          </div>) : <div />}
        {showMap ? (
          <div className={`${styles.mapView} ${!showMap && styles.hide}`}>
            <GoogleMapSearch
              listingsOfGroup={listingsOfGroup}
              markersData={listingsMapResult}
              onActiveListing={onActiveListing}
              currentActive={activeListing}
              activeListing={activeListingResults}
              onListingSelected={onListingSelected}
              onMarkerGroupListingClick={onMarkerGroupListingClick}
              onMapViewModeChanged={onMapViewModeChanged}
            />
          </div>
        ) : <div /> }
      </div>
    </Mobile>
  </div>
);

SearchResultComponents.propTypes = {
  foundItems: PropTypes.number.isRequired,
  listings: PropTypes.array.isRequired,
  listingsOfGroup: PropTypes.array.isRequired,
  facetSearchItems: PropTypes.array.isRequired,
  pageSize: PropTypes.number.isRequired,
  currentPage: PropTypes.number,
  activeListing: PropTypes.any,
  showMap: PropTypes.bool.isRequired,
  onViewModeChanged: PropTypes.func.isRequired,
  onMapViewModeChanged: PropTypes.func.isRequired,
  onMarkerGroupListingClick: PropTypes.func.isRequired,
  onPageChanged: PropTypes.func.isRequired,
  onListingSelected: PropTypes.func.isRequired,
  listingsMapResult: PropTypes.array.isRequired,
  onListingSaved: PropTypes.func.isRequired,
  onSortRusultChanged: PropTypes.func.isRequired,
  onActiveListing: PropTypes.func,
  activeListingResults: PropTypes.any,
  viewMode: PropTypes.string,
  t: PropTypes.func.isRequired,
};

SearchResultComponents.defaultProps = {
  activeListing: {},
  currentPage: 1,
  activeListingResults: {},
  onActiveListing: noop,
  viewMode: ViewOptions.List,
};

export default translate(['searchResult'])(SearchResultComponents);
