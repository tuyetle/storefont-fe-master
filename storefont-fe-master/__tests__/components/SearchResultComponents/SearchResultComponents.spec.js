import React from 'react';
import renderer from 'react-test-renderer';
import ViewOptions from '~/const/ViewOptions';
import SearchResultComponents from '~/components/SearchResultComponents/SearchResultComponents';

describe('(Component) SearchResultComponents', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      foundItems: 1,
      listings: [{
        id: 1,
        subTitle: 'TOP',
        title: 'Chung Cư Tây Hồ RiverView ',
        description: 'Căn hộ có thiết kế hiện đại và đa dạng',
        showImage: true,
        address: 'An Dương Vương,Phường Phú Thượng,Quận Tây Hồ',
        area: 189,
        bathRoom: 3,
        bedRoom: 4,
        price: 12340000000,
        saved: true,
        slug: 'abc-def-123',
        atrributes: {
          bedroom: '2',
          bathroom: '1',
          guestroom: '1',
        },
        center: {
          lat: '21.08772',
          lng: '105.80664',
        },
        img: 'http://www.muabannhadat.vn//uploads/images/005/663/592/hY-y969gzUyLyAIGAvxdRQ_853.jpg',
      }],
      listingsOfGroup: [],
      facetSearchItems: [],
      pageSize: 10,
      currentPage: 1,
      activeListing: {},
      showMap: false,
      onViewModeChanged() {},
      onMapViewModeChanged() {},
      onMarkerGroupListingClick() {},
      onPageChanged() {},
      onListingSelected() {},
      listingsMapResult: [],
      onListingSaved() {},
      onSortRusultChanged() {},
      onActiveListing() {},
      activeListingResults: {},
      viewMode: ViewOptions.Grid,
    };

    component = <SearchResultComponents {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
