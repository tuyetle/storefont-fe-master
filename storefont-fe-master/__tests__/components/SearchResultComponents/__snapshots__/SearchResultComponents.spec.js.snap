// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) SearchResultComponents should render conrectly 1`] = `
<div
  className="css-w7lyt2"
/>
`;
